src/api/main.py

```
@@ -303,6 +346,21 @@ def get_shots(db: Session = Depends(get_db), params: QueryParams = Depends()):
     return paginate(db, query)
 
 
+@app.post("/json/shots", description="Post data to shot table")
+def post_shots(
+    shot_data: list[dict],
+    db: Session = Depends(get_db),
+    _: HTTPBasicCredentials = Depends(authenticate_user_by_role),
+):
+    try:
+        engine = db.get_bind()
+        df = pd.DataFrame(shot_data)
+        df.to_sql("shots", engine, if_exists="append", index=False)
+        return shot_data
+    except Exception as e:
+        raise HTTPException(status_code=400, detail=f"Error:{str(e)}")
```

Why are you masking the real exception and returning an HTTPException? Could this not become hard to debug if the exception is actually coming from pandas?